tactics/autorewrite.ml

```diff
@@ -38,11 +38,11 @@ let subst_hint subst hint =
 
 module HintIdent =
 struct
-  type t = int * rew_rule
+  type t = KerName.t * rew_rule
 
-  let compare (i, t) (j, t') = i - j
+  let compare (kn, t) (kn', t') = KerName.compare kn kn'
```

Comment:
Is comparison ever used for sorting?

Reply:
It probably affects the order in which rewrite hints are tried, if several match. That's probably the root of the issue.